File: 2019/01-1.py
import math

print('Mass       Fuel')
print('------------------------')

fuel_total = 0
with open('01.txt') as fp:
    for line in fp:
        mass = int(line.strip())
        fuel = math.floor(mass / 3) - 2
        fuel_total += fuel
        print(f'{str(mass).ljust(10, " ")} {fuel} {fuel_total}')

print('------------------------')
print(f'Total fuel: {fuel_total}')

File: 2019/01-1.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(void) {
    FILE * fp;
    char * line = NULL;
    size_t len = 0;
    ssize_t read;

	int mass = 0;
	int fuel_total = 0;

    fp = fopen("01.txt", "r");

    while ((read = getline(&line, &len, fp)) != -1) {
		sscanf(line, "%d", &mass);
		fuel_total += floor((mass / 3) - 2);
		printf("%d : %d\n", mass, fuel_total);
    }

    fclose(fp);
    if (line) free(line);
    exit(0);
}
